src/Uno.UI/UI/Xaml/Media/Animation/Animators/GPUFloatValueAnimator.iOSmacOS.cs

```diff
@@ -417,22 +431,66 @@ private NSValue ToCASkewTransform(float angleX, float angleY)
 			return NSValue.FromCATransform3D(CATransform3D.MakeFromAffine(matrix));
 		}
 
+		private void OnCoreWindowVisibilityChanged(CoreWindow sender, VisibilityChangedEventArgs args)
+		{
+			// note: There is no guarantee on which is called first between this (didEnterBackgroundNotification)
+			// and FinalizeAnimation (animationDidStop).
+			if (!args.Visible)
+			{
+				if (IsRunning || _coreStoppedNotFinished)
+				{
+					Pause();
+					_isPausedInBackground = true;
+				}
+			}
+			else
+			{
+				if (_isPausedInBackground)
+				{
+					if (_coreAnimation is { })
+					{
+						Resume();
+					}
+					else
+					{
+						Start();
+					}
+				}
+			}
+		}
+
 		private void FinalizeAnimation(UnoCoreAnimation.CompletedInfo completedInfo)
 		{
 			if (this.Log().IsEnabled(Uno.Foundation.Logging.LogLevel.Debug))
 			{
 				this.Log().DebugFormat("Finalizing animation for GPU Float value animator on property {0}.", _bindingPath.LastOrDefault().PropertyName);
 			}
 
-			if (_valueAnimator?.IsRunning ?? false)
+			var wasRunning = _valueAnimator?.IsRunning ?? false;
+			if (wasRunning)
 			{
 				_valueAnimator.Cancel();
 			}
 
 			switch(completedInfo)
 			{
-				case UnoCoreAnimation.CompletedInfo.Sucesss: AnimationEnd?.Invoke(this, EventArgs.Empty); break;
-				case UnoCoreAnimation.CompletedInfo.Error: AnimationFailed?.Invoke(this, EventArgs.Empty); break;
+				case UnoCoreAnimation.CompletedInfo.Success:
+					AnimationEnd?.Invoke(this, EventArgs.Empty);
+					break;
+
+				case UnoCoreAnimation.CompletedInfo.Error:
+					// ref: https://developer.apple.com/documentation/quartzcore/caanimationdelegate/2097259-animationdidstop
+					// This callback from animationDidStop:finished with the finished=false (which is translated to Error here)
+					// means that the animation have ended because "it has been removed from the layer it is attached to."
+					// It could mean it was explicitly CoreAnimation::StopAnimation'd by us, or killed by the os as the app is background'd.
+					// Since we don't know which case it is, we are just setting another flag to be later confirmed in OnCoreWindowVisibilityChanged.
+					AnimationFailed?.Invoke(this, EventArgs.Empty);
+					if (wasRunning)
+					{
+						_coreStoppedNotFinished = true;
```

Comment:
for the case here, idk if we should save the time when this line happens,
and verify the app is background'd with certain reasonable time window when confirming that it indeed _`_isPausedInBackground`_?

Reply:
@dr1rrb

Reply:
I'm not sure to see the case you are trying to catch by adding such a delay ? IMO for now we should keep it as simple as possible and avoid weird hard coded delay like this.

Regarding your comment I would assume that we should not raise the `AnimationFailed` event in that case.

Reply:
> The animation may have ended because it has completed its active duration or because it has been removed from the layer it is attached to. flag is true if the animation reached the end of its duration without being removed.

I feel it make sense to keep that event proc, as the animation was forcibly removed.
My question was since there can be multiple reasons on why the animation was removed:
1. The app was put into background
2. The animation paused/crashed somehow (lets just assume)

Just \#1 alone is fine. The comment was for a \#2 followed by \#1, which could lead to some confusion/mishandling.